packages/plugin-react/src/index.ts

```diff
@@ -141,11 +142,20 @@ export default function viteReact(opts: Options = {}): PluginOption[] {
           )
       })
 
-      if (typeof opts.babel !== 'function') {
-        runPluginOverrides(config, babelOptions)
-      }
+      runPluginOverrides = (babelOptions) => {
+        const hooks = config.plugins
+          .map((plugin) => plugin.api?.reactBabel)
+          .filter(Boolean) as ReactBabelHook[]
```

Comment:
I've added a small optimization that avoids iterating over plugins without an `api.reactBabel` hook every single time